src/cargo/ops/cargo_rustc/fingerprint.rs

```diff
@@ -173,25 +175,36 @@ fn deserialize_deps<'de, D>(d: D) -> Result<Vec<(String, Arc<Fingerprint>)>, D::
 enum LocalFingerprint {
     Precalculated(String),
     MtimeBased(MtimeSlot, PathBuf),
+    EnvBased(VarSlot, String),
 }
 
 struct MtimeSlot(Mutex<Option<FileTime>>);
+struct VarSlot(Mutex<Option<String>>);
```

Comment:
Hm, why do we need need to compute env-based fingerprints lazily? That is, why do we need `Mutex` here?

For `MtimeSlot` is needed because filesystem state will change after cargo is run, so we need to fetch mtime late.

But `env::var(var)` does not change, so we can eagerly calculate it.